packages/apps/desktop-viewer-test/src/backend/main.ts

```diff
@@ -29,6 +29,10 @@ const viewerMain = async () => {
   Logger.setLevel(AppLoggerCategory.Backend, LogLevel.Info);
 
   const clientId = getAppEnvVar("CLIENT_ID") ?? "";
+  assert(
```

Comment:
@calebmshafer @Carl-Hinkle @kckst8 

Any objections?

Reply:
I haven't looked through the rest of the template but wouldn't this be caught the same way we catch client ids in web now, via the environment vars?

Reply:
afaik, we don't actually check what kind of client is being used in web either. 
we noticed the other day that you could theoretically use a spa client id with the desktop viewer atm, and wanted to prevent that and made sure ppl were using the right client type for their app.

Reply:
whats the harm in using a spa-**** client? Once the desktop viewer has proper local briefcase support, I assume native-xxx will be required, but for now its just snapshots or online.

Reply:
No harm atm, just wanted to be pro-active about this. As long as later on when we do require a desktop client id, the user knows they need one, and not one for a spa

Reply:
@calebmshafer any thoughts on this? Guess I'm not sure what the difference is outside of knowing that they registered a desktop app for internal reporting. Guess if that is the case we should enforce it to have an accurate representation.

Reply:
There is an explicit different between the two and I'm fairly certain using a spa client wouldn't work since technically that uses Auth Code flow and the native clients are registered as Auth Code + PKCE support.

That being said, I do not think we really need the assert as we don't valid that the web viewer is a spa so I see no reason to do the same for Desktop.  If our docs do not lead you down the correct path, then those need to be fixed rather than using this as a blocker..